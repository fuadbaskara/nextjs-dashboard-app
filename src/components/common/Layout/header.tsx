/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import {} from 'react-icons/';
import Image from 'next/future/image';
import Link from 'next/link';

function Header() {
  return (
    <header className="fixed top-0 z-10 block w-full bg-white p-4 shadow shadow-stone-200">
      <div className="w-full px-8">
        <Link href="/">
          <Image
            src="/images/ajaib-logo.png"
            width={320 * 0.48}
            height={118 * 0.48}
            quality={30}
            alt="ajaib-logo"
            className="h-auto w-20 w-[154px]"
          />
        </Link>
      </div>
    </header>
  );
}

export default Header;
