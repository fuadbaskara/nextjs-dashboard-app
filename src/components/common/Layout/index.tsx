import * as React from 'react';
import Header from './header';

export type LayoutProps = {
  children?: React.ReactNode;
};

function Layout({ children }: LayoutProps) {
  return (
    <>
      <Header />
      <div className="mt-20 block bg-slate-100">
        <div className="flex flex-1 flex-col">
          <div className="container mx-auto space-y-4 py-6">{children}</div>
        </div>
      </div>
    </>
  );
}

Layout.defaultProps = {
  children: null,
};

export default Layout;
