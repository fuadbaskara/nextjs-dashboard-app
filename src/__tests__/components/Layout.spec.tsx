/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { screen } from '@testing-library/react';
import { render } from '../test-utils';
import Layout from '@/components/common/Layout';

const renderLayout = () =>
  render(
    <Layout>
      <div>Some Component</div>
    </Layout>,
  );

describe('Layout Wrapper Component', () => {
  test('renders correctly with simple children component:', () => {
    const { asFragment } = renderLayout();
    expect(screen.getByText('Some Component')).toBeInTheDocument();
    expect(document.querySelector('img')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
