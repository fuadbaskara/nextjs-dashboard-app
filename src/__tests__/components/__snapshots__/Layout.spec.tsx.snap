// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Wrapper Component renders correctly with simple children component: 1`] = `
<DocumentFragment>
  <header
    class="fixed top-0 z-10 block w-full bg-white p-4 shadow shadow-stone-200"
  >
    <div
      class="w-full px-8"
    >
      <img
        alt="logo"
        class="h-auto w-20 w-[154px]"
        data-nimg="future"
        decoding="async"
        height="56.64"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogo.png&w=384&q=30"
        srcset="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=30 1x, /_next/image?url=%2Fimages%2Flogo.png&w=384&q=30 2x"
        style="color: transparent;"
        width="153.6"
      />
    </div>
  </header>
  <div
    class="mt-20 block bg-slate-100"
  >
    <div
      class="flex w-full flex-1 flex-col"
    >
      <div
        class="container mx-auto space-y-4 py-6 px-4"
      >
        <div>
          Some Component
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
